### PR TITLE
Memleak fixes for pbs_sched

### PR DESCRIPTION
**Pull request by arungrover**
#### Describe Bug or Feature
memleak fixes for pbs scheduler

#### Describe Your Change
[mem_leaks.docx](https://github.com/openpbs/openpbs/files/5676543/mem_leaks.docx)

#### Link to Design Doc
NA

#### Attach Test and Valgrind Logs/Output
[valgrind.log](https://github.com/openpbs/openpbs/files/5681444/valgrind.log)
None of the leaks mentioned in the other document is present in this log

### CODE REVIEW

File: src/scheduler/parse.cpp [hunk 1]
```diff
@@ -876,6 +876,15 @@ init_config()
 			free(conf.dynamic_res[i].script_name);
 		}
 	}
+	/* clear peer queues */
+	if (conf.peer_queues[0].local_queue != NULL) {
+		int i;
+		for (i = 0; i < NUM_PEERS; i++) {
```
Comment (vstumpf) [inline src/scheduler/parse.cpp:882]: you can declare `i` in the for loop if you want c:

File: src/scheduler/job_info.cpp [hunk 2]
```diff
@@ -1051,6 +1051,9 @@ query_jobs(status *policy, int pbs_sd, queue_info *qinfo, resource_resv **pjobs,
 
 		if (tdata->error || tdata->oarr == NULL) {
 			pbs_statfree(jobs);
+			if (tdata->oarr != NULL)
```
Comment (vstumpf) [inline src/scheduler/job_info.cpp:1054]: You don't _need_ this if statement, but it's ok

**Review by vstumpf — CHANGES_REQUESTED** [review]
Just some comments

File: src/scheduler/parse.cpp (hunk 1 above)
Comment (arungrover) [reply src/scheduler/parse.cpp:882]: creature of habit :)

File: src/scheduler/job_info.cpp (hunk 2 above)
Comment (arungrover) [reply src/scheduler/job_info.cpp:1054]: That's true, I'll remove it.

**Review by bhroam — APPROVED** [review]
LGTM

**Review by vstumpf — APPROVED** [review]
LGTM

**Pull request merged**